api/src/main/java/com/datatorrent/api/Component.java

```diff
@@ -116,4 +116,13 @@ public boolean equals(Object obj)
 
   }
 
+  /**
+   * Marker interface for the Node in the DAG. Any object which can be added as a Node in the DAG
+   * needs to implement this interface.
+   */
+  interface Vertex
```

Comment:
How about moving it into DAG as this is specific to DAG. Component is more general and is implemented by artifacts that are not part of DAG such as security components.